RecentActivity/src/org/sleuthkit/autopsy/recentactivity/BinaryCookieReader.java

```diff
@@ -0,0 +1,429 @@
+    private class CookiePage implements Iterable<Cookie> {
+
+        int[] cookieOffSets;
+        ByteBuffer pageBuffer;
+
+        /**
+         * Setup the CookiePage object. Calidates that the page bytes are in the
+         * correct format by checking for the header value of 0x0100
+         *
+         * @param page
+         * @throws IOException
+         */
+        CookiePage(byte[] page) throws IOException {
+            if (page == null || page.length == 0) {
+                throw new IllegalArgumentException("Invalid value for page passes to CookiePage constructor"); //NON-NLS
+            }
+
+            pageBuffer = ByteBuffer.wrap(page);
+
+            if (pageBuffer.getInt() != PAGE_HEADER_VALUE) {
+                pageBuffer = null;
+                throw new IOException("Invalid file format, bad page head value found"); //NON-NLS
+            }
+
+            pageBuffer.order(ByteOrder.LITTLE_ENDIAN);
+            int count = pageBuffer.getInt();
+            cookieOffSets = new int[count];
+
+            for (int cnt = 0; cnt < count; cnt++) {
+                cookieOffSets[cnt] = pageBuffer.getInt();
+            }
+
+            pageBuffer.getInt(); // All 0, not needed
+        }
+
+        /**
+         * Returns in instance of a CookieIterator
+         *
+         * @return CookieIterator
+         */
+        @Override
+        public Iterator<Cookie> iterator() {
+            return new CookieIterator();
+        }
+
+        /**
+         * Implements Iterator to iterate over the cookies in the page
+         */
+        private class CookieIterator implements Iterator<Cookie> {
+
+            int index = 0;
+
+            /**
+             * Checks to see if there are more cookies
+             *
+             * @return True if there are more cookies, false if there are not
+             */
+            @Override
+            public boolean hasNext() {
+                if (pageBuffer == null) {
+                    return false;
+                }
+
+                return index < cookieOffSets.length;
+            }
+
+            /**
+             * Gets the next cookie from the page
+             *
+             * @return Next cookie
+             */
+            @Override
+            public Cookie next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+
+                int offset = cookieOffSets[index];
+                int size = pageBuffer.getInt(offset);
+                byte[] cookieBytes = new byte[size];
+                pageBuffer.get(cookieBytes, 0, size);
+                index++;
+
+                return new Cookie(cookieBytes);
+            }
+        }
+    }
+
+    /**
+     * Represents an instance of a cookie from the binarycookie file
+     */
+    public class Cookie {
+
+        private final static int COOKIE_HEAD_SKIP = 16;
+
+        private final Double expirationDate;
+        private final Double creationDate;
```

Comment:
Do Double objects need to be used here instead of double primitives?

Reply:
Changed them to primitives.